### PR TITLE
Fix lack of MXNet plugin docs generation

### PR DESCRIPTION
**Pull request by JanuszL**
Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It fixes lack of MXNet plugin docs generation

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     removes MXNet from Sphinx mocks due to sphinx-doc/sphinx#5995
     fixes warnings from docs generation
 - Affected modules and functionalities:
     documentation
 - Key points relevant for the review:
     NA
 - Validation and testing:
     CI and checking the generated docs
 - Documentation (including examples):
     Docs were updated

**JIRA TASK**: *[NA]*

### CODE REVIEW

File: dali/operators/reader/numpy_reader_op.cc [hunk 1]
```diff
@@ -58,10 +58,10 @@ directories with numpy files in them.)code",
       DALI_STRING)
   .AddOptionalArg("file_filter",
       R"code(If specified, the string will be interpreted as glob string to filter
-the list of files in the sub-directories of `file_root`.)code", "*.npy")
+the list of files in the sub-directories of `file_root`.)code", "\\*.npy")
```
Comment (JanuszL) [inline dali/operators/reader/numpy_reader_op.cc:61]: Fix

File: dali/python/nvidia/dali/external_source.py [hunk 2]
```diff
@@ -199,25 +200,25 @@ class ExternalSource():
...
-``no_copy` : Whether DALI should copy the buffer when feed_input is called
+`no_copy` : Whether DALI should copy the buffer when feed_input is called
     If True, DALI passes the user memory directly to the Pipeline, instead of copying.
     It is the user's responsibility to keep the buffer alive and unmodified
     until it is consumed by the pipeline.
 
     The buffer can be modified or freed again after the relevant iteration output has been consumed.
-    Effectively, it happens after ``prefetch_queue_depth`` or ``cpu_queue_depth * gpu_queue_depth``
     (when they are not equal) iterations following the``feed_input`` call.
+    Effectively, it happens after ``prefetch_queue_depth`` or ``cpu_queue_depth \* gpu_queue_depth``
```
Comment (JanuszL) [inline dali/python/nvidia/dali/external_source.py:213]: Fix

File: docs/index.rst [hunk 3]
```diff
@@ -11,8 +11,8 @@ NVIDIA DALI documentation
    .. warning::
       You are currently viewing unstable developer preview of the documentation.
       To see the documentation for the latest stable release, refer to:
-        * `Release Notes <https://docs.nvidia.com/deeplearning/dali/release-notes/index.html>`_
-        * `Developer Guide <https://docs.nvidia.com/deeplearning/dali/user-guide/docs/index.html>`_ (stable version of this page)
+      * `Release Notes <https://docs.nvidia.com/deeplearning/dali/release-notes/index.html>`_
```
Comment (JanuszL) [inline docs/index.rst:14]: Fix

Comment (JanuszL) [reply docs/index.rst:14]: Done

File: dali/python/nvidia/dali/external_source.py (hunk 2 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/external_source.py:213]: Done

File: dali/operators/reader/numpy_reader_op.cc (hunk 1 above)
Comment (JanuszL) [reply dali/operators/reader/numpy_reader_op.cc]: Done

**Comment by JanuszL** [comment]
!build

File: dali/python/nvidia/dali/external_source.py [hunk 4]
```diff
@@ -346,14 +347,17 @@ def external_source(source = None, num_outputs = None, *, cycle = None, name = N
...
 ``pipeline.feed_input(name, data, layout, cuda_stream)`` inside ``pipeline.iter_setup``.
-    In the case of the GPU input, it is the user responsibility to modify the
-    provided GPU memory content only using provided stream (DALI schedules a copy on it
-    and all work is properly queued). If no stream is provided feeding input blocks until the
-    provided memory is copied to the internal buffer
+
+In the case of the GPU input, it is the user responsibility to modify the
+provided GPU memory content only using provided stream (DALI schedules a copy on it
+and all work is properly queued). If no stream is provided feeding input blocks until the
+provided memory is copied to the internal buffer
```
Comment (jantonguirao) [inline dali/python/nvidia/dali/external_source.py:354]:
```suggestion
provided memory is copied to the internal buffer.
```

File: cmake/Utils.cmake [hunk 5]
```diff
@@ -324,7 +324,7 @@ function(build_per_python_lib)
         set(PYTHON_LIB_TARGET_FOR_PYVER "${PYTHON_LIB_ARG_TARGET_NAME}_${PYVER}")
         # check if listed python versions are accesible
         execute_process(
-            COMMAND python${PYVER} -c "print('test', end='')"
+            COMMAND python${PYVER} -c "print('', end='')"
```
Comment (klecki) [inline cmake/Utils.cmake:327]: As you already touched this:
```suggestion
            COMMAND python${PYVER} -c "exit()"
```

File: dali/python/nvidia/dali/external_source.py (hunk 4 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/external_source.py]: Done

File: dali/python/backend_impl.cc [hunk 6]
```diff
@@ -284,8 +284,6 @@ void ExposeTensor(py::module &m) {
       Check if provided python object represent a valid DLPack capsule.
       It returns a tuple of two boolean values: one indicating if this is a valid DLPack object, and the other if the data
 
-      Parameters
-      ----------
```
Comment (klecki) [inline dali/python/backend_impl.cc:288]: I don't see any difference in the resulting docs, what happened here?

File: cmake/Utils.cmake (hunk 5 above)
Comment (JanuszL) [reply cmake/Utils.cmake]: Done

File: dali/python/backend_impl.cc (hunk 6 above)
Comment (JanuszL) [reply dali/python/backend_impl.cc:288]: That is good. Apparently it was not needed, and only sphinx was complaining with:
```
docstring of nvidia.dali.backend.TensorListCPU.__init__:9: WARNING: Unexpected section title.
 Parameters
 ----------
```

**Pull request merged**